docs/extensions/guides/ipc.md

```diff
@@ -0,0 +1,131 @@
+# Inter Process Communication
+
+A Lens Extension can utilize IPC to send information between its `LensRendererExtension` and its `LensMainExtension`.
+This is useful when wanting to communicate directly within your extension.
+For example, if a user logs into a service that your extension is a facade for and `main` needs to know some information so that you can start syncing items to the `Catalog`, this would be a good way to send that information along.
+
+IPC channels are blocked off per extension.
+Meaning that each extension can only communicate with itself.
+
+## Types of IPC
+
+There are two flavours of IPC that are provided:
+
+- Event based
+- Request based
+
+### Event Based IPC
+
+This is the same as an [Event Emitter](https://nodejs.org/api/events.html#events_class_eventemitter) but is not limited to just one Javascript process.
+This is a good option when you need to report that something has happened but you don't need a response.
+
+This is a fully two-way form of communication.
+Both `LensMainExtension` and `LensRendererExtension` can do this sort of IPC.
+
+### Request Based IPC
+
+This is more like a Remote Procedure Call (RPC).
+With this sort of IPC the caller waits for the result from the other side.
+This is accomplished by returning a `Promise<T>` which needs to be `await`-ed.
+
+This is a unidirectional form of communication.
+Only `LensRendererExtension` can initiate this kind of request, and only `LensMainExtension` can and respond this this kind of request.
+
+## Registering IPC Handlers and Listeners
+
+The general terminology is as follows:
+
+- A "handler" is the function that responds to a "Request Based IPC" event.
+- A "listener" is the function that is called when a "Event Based IPC" event is emitted.
+
+To register either a handler or a listener, you should do something like the following:
+
+`main.ts`:
+```typescript
+import { LensMainExtension, Interface, Types, Store } from "@k8slens/extensions";
+import { registerListeners, IpcMain } from "./helpers/main";
+
+export class ExampleExtensionMain extends LensMainExtension {
+  onActivate() {
+    IpcMain.createInstance(this);
+
+    registerListeners();
+  }
+}
+```
+
+This file shows that you need to create an instance of the store to be able to use IPC.
+Lens will automatically clean up that store and all the handlers on deactivation and uninstall.
+
+---
+
+`helpers/main.ts`:
+```typescript
+import { Store } from "@k8slens/extensions";
+
+export class IpcMain extends Store.MainIpcStore {}
+
+function onInitialize(event: Types.IpcMainEvent, id: string) {
+  console.log(`starting to initialize: ${id}`);
+}
+
+export function registerListeners() {
```

Comment:
These functions could be also within `IpcMain`, not sure which one is more fluent for a reader.

Reply:
Oh you mean something like the following?

```ts
export class IpcMain extends Store.MainIpcStore {
  constructor() {
    super();

    this.listenIpc("initialize", onInitialize);
  }
}
```

Reply:
Yes, something like that.

Reply:
I think that is indeed better.

Reply:
Seems more natural.

Reply:
Changed.